app/controllers/users_controller.rb

```diff
@@ -78,6 +78,35 @@ def oauth_github()
     redirect_to "/"
   end
 
+  def oauth_google()
+    code = params[:code]
+    validator = GoogleIDToken::Validator.new
+    jwt = validator.check(code, google_client.id, google_client.id)
+    if jwt
+      google_id = jwt['sub']
+      google_name = jwt['name']
+      email = jwt['email']
+
+      u = User.find_or_create_by(google_id: google_id)
+      if !u.persisted?
+        u.name = google_name
+        u.email = email
+        u.password = SecureRandom.hex
```

Comment:
Just realized; if we're logging in with all of these OAuth services, do we even need a password field?